File: tools/census-harvest/tests/fixtures/ts-mini/src/store.ts
export interface Config {
  maxSize: number;
  region: string;
}

export interface Stats {
  objects: number;
}

export class Store {
  cfg: Config;
  private count: number;

  constructor(cfg: Config) {
    this.cfg = cfg;
    this.count = 0;
  }

  getObject(key: string): Promise<Uint8Array | null> {
    if (key === "") {
      return Promise.resolve(null);
    }
    return Promise.resolve(new Uint8Array([1, 2, 3]));
  }

  putObject(key: string, data: Uint8Array): boolean {
    if (key === "" || data.length === 0) {
      return false;
    }
    this.bump();
    return true;
  }

  stats(): Stats {
    return { objects: this.count };
  }

  protected bump(): void {
    this.count += 1;
  }
}

File: plugins/igr/tools/census-harvest/tests/fixtures/go-mini/store/store.go
package store

type Config struct {
	MaxSize int
	Region  string
}

type Stats struct {
	Objects uint64
}

type Store struct {
	Cfg   Config
	count uint64
}

func New(cfg Config) *Store {
	return &Store{Cfg: cfg}
}

func (s *Store) GetObject(key string) ([]byte, error) {
	if key == "" {
		return nil, errEmpty
	}
	return []byte{1, 2, 3}, nil
}

func (s *Store) PutObject(key string, data []byte) error {
	if key == "" || len(data) == 0 {
		return errEmpty
	}
	s.bump()
	return nil
}

func (s *Store) Stats() Stats {
	return Stats{Objects: s.count}
}

func (s *Store) bump() {
	s.count++
}
